internal/controller/etcdcluster_controller.go

```diff
@@ -78,13 +78,6 @@ func (r *EtcdClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, err
 	}
 
-	if etcdCluster.Spec.Size == 0 {
-		logger.Info("EtcdCluster size is 0..Skipping next steps")
-		return ctrl.Result{}, nil
-	}
```

Comment:
Suggest to keep this no matter we add the restriction (minimum value of cluster size), the controller should be correct by itself. It's also harmless